app/models/group_order_invoice.rb

```diff
@@ -0,0 +1,60 @@
+class GroupOrderInvoice < ApplicationRecord
+
+  belongs_to :group_order
+  validates_presence_of :group_order
+  validates_uniqueness_of :invoice_number
+  validate :tax_number_set
+  after_initialize :init, unless: :persisted?
+
+  def generate_invoice_number(count)
+    trailing_number = count.to_s.rjust(4, '0')
+    unless  GroupOrderInvoice.find_by(invoice_number: Time.now.strftime("%Y%m%d") + trailing_number)
+      Time.now.strftime("%Y%m%d") + trailing_number
+    else
+      generate_invoice_number(count.to_i + 1)
+    end
+  end
+
+  def tax_number_set
+    if FoodsoftConfig[:contact][:tax_number].blank?
+      errors.add(:group_order_invoice, "Keine Steuernummer in FoodsoftConfig :contact gesetzt")
+    end
+  end
+
+  def init
+    self.invoice_number = generate_invoice_number(1) unless self.invoice_number
```

Comment:
please use a `before_validation` hook instead (see e.g. [`set_password` in the `User` model](https://github.com/foodcoops/foodsoft/blob/f7f8774c433d58bcdf0c03fe93a7967fcf0e6437/app/models/user.rb#L122-L127))